app/src/main/java/com/digitalhouse/dhwallet/adapter/PaymentAdapter.kt

```diff
@@ -0,0 +1,109 @@
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.digitalhouse.dhwallet.R
+import com.digitalhouse.dhwallet.model.GroupTransaction
+import com.digitalhouse.dhwallet.model.GroupType
+
+private const val MENU = 0
+private const val NAVTITLE = 1
+private const val CONTENT = 2
+
+
+class PaymentAdapter(
+    private val items: List<GroupTransaction>
+    ): RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        val inflator = LayoutInflater.from(parent.context)
+
+        if(viewType == MENU){
+            return PaymentMenuViewHolder(
+                inflator.inflate(
+                    R.layout.payment_menu,
+                    parent,
+                    false
+                )
+            )
+        }
+
+        if(viewType == NAVTITLE){
+            return PaymentTitleViewHolder(
+                inflator.inflate(
+                    R.layout.nav_title_transaction,
+                    parent,
+                    false
+                )
+            )
+        }
+
+        return PaymentContentViewHolder(
+            inflator.inflate(
+                R.layout.item_transaction,
+                parent,
+                false
+            )
+        )
+    }
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        when(holder){
+            is PaymentTitleViewHolder -> holder.bindNavPay(items[position])
+            is PaymentContentViewHolder -> holder.bind(items[position])
+        }
+    }
+
+
+    override fun getItemViewType(position: Int): Int {
+
+        if(items[position].type == GroupType.NAVTITLE){
+            return NAVTITLE
+        }
+
+        if(items[position].type == GroupType.CONTENT){
+            return CONTENT
+        }
+
+        return MENU
+    }
+
+
+    override fun getItemCount(): Int = items.size
+}
+
+
+class PaymentContentViewHolder(view: View): RecyclerView.ViewHolder(view){
+
+    private val image : ImageView = view.findViewById(R.id.item_transaction_image)
+    private val title : TextView = view.findViewById(R.id.item_transaction_title)
+    private val subtitle : TextView = view.findViewById(R.id.item_transaction_subtitle)
+    private val value : TextView = view.findViewById(R.id.item_transaction_value)
+
+    fun bind(item: GroupTransaction){
+        Glide.with(image.context).load(item.image).circleCrop().into(image);
+        title.text = item.title
+        subtitle.text = item.subtitle
+        value.text = ""
+    }
+}
+
+
+class PaymentTitleViewHolder(view: View,) : RecyclerView.ViewHolder(view) {
+    private val navTitle: TextView = view.findViewById(R.id.nav_header_transaction_title)
+    private val navIcon: ImageView = view.findViewById(R.id.nav_header_transaction_image)
+
+    fun bindNavPay(item: GroupTransaction){
+        navTitle.text = item.title
+        navIcon.isVisible = false
```

Comment:
Se não me engano a visibilidade do `nav_header_transaction_image` por padrão já é gone no xml, se não for a minha recomendação é que coloque o `gone` no xml, e aí vc não vai precisar forçar ele sempre aqui com o false

Reply:
Como eu reaproveitei a view da transaction e ela tem o icone no final, usei essa estrategia para sumir com o icone e reaproveitar a view, se eu colocar gone no xml o icone desaparece em ambas as situações, e não somente em uma delas como o esperado.